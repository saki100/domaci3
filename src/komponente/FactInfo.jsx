import React from 'react'
import { Link } from 'react-router-dom'

function FactInfo({oneFact}) {
  return (
    <div class="card text-center">
        <div class="card-header">
        <img src="https://static.vecteezy.com/system/resources/previews/009/665/304/original/cute-kitty-cat-head-cartoon-element-free-png.png" className="card-img-top" alt="cat"></img>
            Sve činjenice preuzete su pomoću javnog API-ja
        </div>
        <div class="card-body">
            <h5 class="card-title">Fact Info</h5>
            <p class="card-text">{oneFact.fact}</p>
            <p class="card-text">Broj karaktera: {oneFact.length}</p>
            <Link to="/">
             <button class="btn btn-primary">Nazad</button>
            </Link>
        </div>
        <div class="card-footer text-muted">
            <a href="https://catfact.ninja/#/Facts/getFacts">Link ka API dokumentaciji</a>
        </div>
    </div>
  )
}

export default FactInfo
