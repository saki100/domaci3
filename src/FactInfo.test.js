import React from 'react';
import { render, screen } from '@testing-library/react';
import { BrowserRouter } from 'react-router-dom';
import FactInfo from './komponente/FactInfo';

describe('FactInfo component', () => {
  const mockFact = {
    fact: 'This is a test fact.',
    length: 20,
  };

  test('renders fact information correctly', () => {
    render(
      <BrowserRouter>
        <FactInfo oneFact={mockFact} />
      </BrowserRouter>
    );

    const factText = screen.getByText(/This is a test fact./i);
    expect(factText).toBeInTheDocument();

    const lengthText = screen.getByText(/Broj karaktera: 20/i);
    expect(lengthText).toBeInTheDocument();
  });

  test('renders "Nazad" button', () => {
    render(
      <BrowserRouter>
        <FactInfo oneFact={mockFact} />
      </BrowserRouter>
    );

    const backButton = screen.getByText(/Nazad/i);
    expect(backButton).toBeInTheDocument();
  });

  test('renders API link', () => {
    render(
      <BrowserRouter>
        <FactInfo oneFact={mockFact} />
      </BrowserRouter>
    );

    const apiLink = screen.getByText(/Link ka API dokumentaciji/i);
    expect(apiLink).toBeInTheDocument();
  });

  test('matches snapshot', () => {
    const { asFragment } = render(
      <BrowserRouter>
        <FactInfo oneFact={mockFact} />
      </BrowserRouter>
    );

    expect(asFragment()).toMatchSnapshot();
  });
});
