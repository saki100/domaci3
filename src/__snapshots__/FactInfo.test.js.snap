// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FactInfo component matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="card text-center"
  >
    <div
      class="card-header"
    >
      <img
        alt="cat"
        class="card-img-top"
        src="https://static.vecteezy.com/system/resources/previews/009/665/304/original/cute-kitty-cat-head-cartoon-element-free-png.png"
      />
      Sve činjenice preuzete su pomoću javnog API-ja
    </div>
    <div
      class="card-body"
    >
      <h5
        class="card-title"
      >
        Fact Info
      </h5>
      <p
        class="card-text"
      >
        This is a test fact.
      </p>
      <p
        class="card-text"
      >
        Broj karaktera: 20
      </p>
      <a
        href="/"
      >
        <button
          class="btn btn-primary"
        >
          Nazad
        </button>
      </a>
    </div>
    <div
      class="card-footer text-muted"
    >
      <a
        href="https://catfact.ninja/#/Facts/getFacts"
      >
        Link ka API dokumentaciji
      </a>
    </div>
  </div>
</DocumentFragment>
`;
